src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/on_vr_language_change_notification.cc

```diff
@@ -78,34 +78,42 @@ void OnVRLanguageChangeNotification::Run() {
   (*message_)[strings::params][strings::function_id] =
       static_cast<int32_t>(mobile_apis::FunctionID::OnLanguageChangeID);
 
-  const auto applications = application_manager_.applications().GetData();
-  for (auto app : applications) {
-    if (!app->IsRegistered()) {
-      SDL_LOG_DEBUG("Skipping app "
-                    << app->app_id()
-                    << " which has not finished the registration process");
-      continue;
+  std::vector<uint32_t> app_list;
+  std::vector<uint32_t> to_unregister;
+
+  {
+    const ApplicationSet& accessor =
+        application_manager_.applications().GetData();
+    auto message_language =
+        (*message_)[strings::msg_params][strings::language].asInt();
+
+    for (auto app : accessor) {
+      auto app_id = app->app_id();
+      app_list.push_back(app_id);
+      if (app->language() != message_language) {
+        to_unregister.push_back(app_id);
+        application_manager_.state_controller().SetRegularState(
+            app,
+            mobile_apis::PredefinedWindows::DEFAULT_WINDOW,
+            mobile_apis::HMILevel::HMI_NONE,
+            false);
```

Comment:
Hmmm, seems very odd that we would deactivate the app before unregistering it specifically in the `VR` case. I do realize that this is behavior carried over from the original implementation, but I wonder if it's really necessary